### PR TITLE
CC-13309: Sanitize AuthenticationException error messages

### PR DESCRIPTION
**Pull request by ncliang**

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java [hunk 1]
```diff
@@ -102,10 +101,11 @@ public Config validate(Map<String, String> connectorConfigs) {
...
                 connection.execute("SELECT version()");
                 logger.info("Successfully tested connection for {} with user '{}'", connection.connectionString(),
-                    connection.username());
-            } catch (Throwable e) {
+                        connection.username());
+            }
+            catch (Throwable e) {
                 logger.info("Failed testing connection for {} with user '{}'", config.jdbcConfig(),
-                    userValue);
+                        userValue);
```
Comment (C0urante) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:108]: Are these changes necessary? Appear to be whitespace-only.

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorTest.java [hunk 2]
```diff
@@ -6,40 +6,39 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
 import org.apache.kafka.common.config.Config;
 import org.apache.kafka.common.config.ConfigValue;
 import org.junit.Before;
 import org.junit.Test;
 
-import java.util.Map;
+public class PostgresConnectorTest {
```
Comment (C0urante) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorTest.java:20]: Are there any functional changes in this file? AFAICT there's just some shuffling of import and method order.

**Review by C0urante — APPROVED** [review]
LGTM, a few questions but feel free to address at your discretion.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java (hunk 1 above)
Comment (ncliang) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:108]: Yeah... `mvn clean test` auto-formatted these files and I just left them. Probably debezium enforces a slightly different code style.

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorTest.java (hunk 2 above)
Comment (ncliang) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorTest.java:20]: This was a test that depended on guava, which was present in `v1.2.2` before when it was introduced, but removed in `v1.3.1` . I changed the test to remove the use of ImmutableMap to get it to compile. That is all.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java [hunk 3]
```diff
@@ -102,10 +101,11 @@ public Config validate(Map<String, String> connectorConfigs) {
             try (PostgresConnection connection = new PostgresConnection(config.jdbcConfig())) {
                 connection.execute("SELECT version()");
                 logger.info("Successfully tested connection for {} with user '{}'", connection.connectionString(),
-                    connection.username());
-            } catch (Throwable e) {
+                        connection.username());
```
Comment (Cyril-Engels) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:104]: Indent seems off

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java (hunk 1 above)
Comment (Cyril-Engels) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:108]: Here too

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java [hunk 4]
```diff
@@ -447,6 +448,15 @@ protected void doStart() {
         }
     }
 
+    static AuthenticationException sanitizeAuthenticationException(AuthenticationException e) {
+        String sanitizedError = e.getMessage()
+                .replaceAll("[^\\P{Cc}\t\r\n]", "");
+        AuthenticationException sanitized = new AuthenticationException(sanitizedError,
+                e.getErrorCode(), e.getSqlState());
+        sanitized.setStackTrace(e.getStackTrace());
```
Comment (Cyril-Engels) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java:456]: Can we assume the unrecognized control characters only appear in error message but not stack trace?

**Review by Cyril-Engels — APPROVED** [review]
LGTM
